packages/venia-concept/src/components/ProductFullDetail/ProductFullDetail.js

```diff
@@ -17,6 +17,9 @@ import findMatchingVariant from 'src/util/findMatchingProductVariant';
 
 const Options = React.lazy(() => import('../ProductOptions'));
 
+const productIsConfigurable = product =>
```

Comment:
I know you grabbed this logic from the existing implementation, but I'm curious if we should leverage `__typename` from the query response instead. Documentation is a little light on why this field is in the response even though we're not explicitly requesting it, but it's there, and seems like the preferred pattern when you're dealing with interfaces. Think this might look cleaner?

`product.__typename === 'ConfigurableProduct'`

Reply:
The reason this appears to be "magic" is because of Apollo, it automatically adds `__typename` to queries for its own caching purposes. We may want to explicitly add it to the query and proptype so that's more clear.

Reply:
Nice! I'll give it a shot - agreed it's a lot nicer.